lib/chef/application/windows_service.rb

```diff
@@ -189,7 +189,11 @@ def run_chef_client
           config_params += " -c #{Chef::Config[:config_file]}" unless  Chef::Config[:config_file].nil?
           config_params += " -L #{Chef::Config[:log_location]}" unless Chef::Config[:log_location] == STDOUT
           # Starts a new process and waits till the process exits
-          result = shell_out("chef-client #{config_params}", :timeout => Chef::Config[:windows_service][:watchdog_timeout])
+          result = shell_out(
+            "chef-client #{config_params}",
+            :timeout => Chef::Config[:windows_service][:watchdog_timeout],
+            :logger => Chef::Log
```

Comment:
@btm your memory is correct (as best I can remember myself, heh). In the olden times, users had a bad time when they set the log location in `client.rb` and then logged in to run chef-client manually for debugging or whatever, because the output would go only to the specified logfile and they wouldn't see anything. So we added a feature to make chef-client also log to STDOUT if it detected a TTY. Nowadays this would be less of a problem because we have the formatters, too, but it could be an issue if you needed to get the debug logs.